### ISSUE TITLE
compiler warnings

### ISSUE THREAD

**Issue by underdarkskies** [RavenProject/Ravencoin#187]
From ubuntu 18 bionic running a native, NON cross-compile build with master branch:
```
sudo apt install -y build-essential libtool autotools-dev automake pkg-config libssl-dev libevent-dev bsdmainutils python3 libminiupnpc-dev libzmq3-dev git libboost-all-dev software-properties-common
sudo apt-get install -y libqt5gui5 libqt5core5a libqt5dbus5 qttools5-dev qttools5-dev-tools libprotobuf-dev protobuf-compiler libqrencode-dev
sudo add-apt-repository ppa:bitcoin/bitcoin
sudo apt-get update
sudo apt-get install libdb4.8-dev libdb4.8++-dev
git clone https://github.com/ravenproject/ravencoin.git
cd ravencoin
sudo ./autogen.sh
./configure
make -j4
sudo make install DESTDIR
```
the compiler warns:
```
./consensus/consensus.h:34:13: warning: ‘fAssetsIsActive’ defined but not used [-Wunused-variable]
 static bool fAssetsIsActive = false;
```
and
```
./assets/assets.h: In copy constructor ‘CAssetsCache::CAssetsCache(const CAssetsCache&)’:
./assets/assets.h:139:5: warning: base class ‘class CAssets’ should be explicitly initialized in the copy constructor [-Wextra]
     CAssetsCache(const CAssetsCache& cache)
```

**Comment by underdarkskies** [comment]
*the build compiles complete fine though, despite the warnings.

**Comment by underdarkskies** [comment]
Side-note, in regards to #183 this was configured without `-fPIC`  flags, infact with NO configure flags at all and all binaries get built with hardened code without having to compensate for a buggy gcc versions :) 

```ubuntu@ubuntu:~/test/usr/local/bin$ scanelf -e ../lib/libravenconsensus.so.0.0.0
 TYPE   STK/REL/PTL FILE
ET_DYN RW- R-- RW- ../lib/libravenconsensus.so.0.0.0
ubuntu@ubuntu:~/test/usr/local/bin$ scanelf -e ./ravend
 TYPE   STK/REL/PTL FILE
ET_DYN RW- R-- RW- ./ravend
ubuntu@ubuntu:~/test/usr/local/bin$ scanelf -e ./raven-cli
 TYPE   STK/REL/PTL FILE
ET_DYN RW- R-- RW- ./raven-cli
ubuntu@ubuntu:~/test/usr/local/bin$ scanelf -e ./raven-qt
 TYPE   STK/REL/PTL FILE
ET_DYN RW- R-- RW- ./raven-qt
```

**Comment by bedri** [comment]
```
./consensus/consensus.h:34:13: warning: ‘fAssetsIsActive’ defined but not used [-Wunused-variable]
 static bool fAssetsIsActive = false;
```

Since fAssetsIsActive defined and initiated in consensus.h however this still gives the warning.
It can be initiated in either a function in consensus.h file (or in any function that will be executed after fAssetsIsActive definition and before any function that uses fAssetsIsActive variable)  or fAssetsIsActive can be initiated in all the other functions that uses it before it's used.
Initiating it with a function in consensus.h will break consensus.h file format and purpose and the second approach will create an unneccessary multi initialization of a global variable.
I had looked for a good way of fixing that warning but couldn't find any yet. Up to now this warning did not harm the execution of the software however it will of course be wise to fix this.

**Comment by bedri** [comment]
As far as I remember this issue had been solved when 2.0.1 was being released. @underdarkskies can you confirm that the issue is solved?
Thank you.

**Comment by cfox** [comment]
^^
